packages/forms/src/directives/validators.ts

```diff
@@ -579,8 +582,14 @@ export class MinLengthValidator implements Validator, OnChanges {
   }
 
   private _createValidator(): void {
-    this._validator = minLengthValidator(
-        typeof this.minlength === 'number' ? this.minlength : parseInt(this.minlength, 10));
+    // Enable validation for any truthy value.
+    // If the value is `0`, we reset validator since it's a noop.
+    if (this.minlength) {
+      this._validator = minLengthValidator(
+          typeof this.minlength === 'number' ? this.minlength : parseInt(this.minlength, 10));
+    } else {  // reset to null in case value is set back to `0`
```

Comment:
I am not sure this comment is necessary.

```suggestion
    } else {
```